lib/apiservers/engine/backends/eventmonitor.go

```diff
@@ -236,8 +238,21 @@ func (p DockerEventPublisher) PublishEvent(event plevents.BaseEvent) {
 			attrs["exitCode"] = code
 			log.Infof("Sending die event for container %s - code: %s", vc.ContainerID, code)
 			actor := CreateContainerEventActorWithAttributes(vc, attrs)
-
 			EventService().Log(containerDieEvent, eventtypes.ContainerEventType, actor)
+
+			// auto-remove if required
+			if vc.HostConfig.AutoRemove {
```

Comment:
I'm not sure if the HostConfig is persisted across - I'm sure that autodeletion isn't recorded directly in the containerVM spec.